vnext/Microsoft.ReactNative/Fabric/FabricUIManagerModule.h

```diff
@@ -0,0 +1,218 @@
+  void replaceExistingNonRootView(double reactTag, double newReactTag) noexcept;
+
+  REACT_METHOD(removeRootView)
+  void removeRootView(double reactTag) noexcept;
+
+  REACT_METHOD(setChildren)
+  void setChildren(double containerTag, winrt::Microsoft::ReactNative::JSValueArray &&reactTags) noexcept;
+
+  REACT_METHOD(manageChildren)
+  void manageChildren(
+      double containerTag,
+      winrt::Microsoft::ReactNative::JSValueArray &&moveFromIndices,
+      winrt::Microsoft::ReactNative::JSValueArray &&moveToIndices,
+      winrt::Microsoft::ReactNative::JSValueArray &&addChildReactTags,
+      winrt::Microsoft::ReactNative::JSValueArray &&addAtIndices,
+      winrt::Microsoft::ReactNative::JSValueArray &&removeAtIndices) noexcept;
+
+  REACT_METHOD(setLayoutAnimationEnabledExperimental)
+  void setLayoutAnimationEnabledExperimental(bool enabled) noexcept;
+
+  REACT_METHOD(sendAccessibilityEvent)
+  void sendAccessibilityEvent(double reactTag, double eventType) noexcept;
+
+  REACT_METHOD(showPopupMenu)
+  void showPopupMenu(
+      double reactTag,
+      winrt::Microsoft::ReactNative::JSValueArray &&items,
+      std::function<void(winrt::Microsoft::ReactNative::JSValue const &)> const &error,
+      std::function<void(winrt::Microsoft::ReactNative::JSValue const &)> const &success) noexcept;
+
+  REACT_METHOD(dismissPopupMenu)
+  void dismissPopupMenu() noexcept;
+
+  void startSurface(
+      facebook::react::IReactRootView *rootview,
+      facebook::react::SurfaceId surfaceId,
+      const std::string &moduleName,
+      const folly::dynamic &initialProps) noexcept;
+
+  const ComponentViewRegistry &GetViewRegistry() const noexcept;
+
+ private:
+  void installFabricUIManager() noexcept;
+  void initiateTransaction(facebook::react::MountingCoordinator::Shared const &mountingCoordinator);
+  void performTransaction(facebook::react::MountingCoordinator::Shared const &mountingCoordinator);
+  void RCTPerformMountInstructions(
+      facebook::react::ShadowViewMutationList const &mutations,
+      // facebook::react::RCTComponentViewRegistry* registry,
+      // facebook::react::RCTMountingTransactionObserverCoordinator& observerCoordinator,
+      facebook::react::SurfaceId surfaceId);
+  void didMountComponentsWithRootTag(facebook::react::SurfaceId surfaceId) noexcept;
+
+  // std::shared_ptr<facebook::react::MessageQueueThread> m_batchingUIMessageQueue;
+  // std::shared_ptr<UIManagerModule> m_module;
+  winrt::Microsoft::ReactNative::ReactContext m_context;
+  std::shared_ptr<facebook::react::Scheduler> m_scheduler;
+  std::mutex m_schedulerMutex; // Protect m_scheduler
+  bool m_collapseDeleteCreateMountingInstructions{false};
+  bool m_disablePreallocateViews{false};
+  bool m_transactionInFlight{false};
+  bool m_followUpTransactionRequired{false};
+
+  ComponentViewRegistry m_registry;
+
+  std::unordered_map<facebook::react::SurfaceId, XamlView> m_surfaceRegistry;
+
+  std::recursive_mutex m_commitMutex;
```

Comment:
Not used?

Curious what scenario has reentrancy here as well.

Reply:
Not sure.  Android seems to have the mutex, but iOS doesn't.  Going to remove it for now.